accepted/0003-interactive-audit-resolver.md

```diff
@@ -0,0 +1,92 @@
+- **postpone** marks the issue as postponed to a timestamp 24h in the future. Instead of ignoring an issue permanently just to make a build pass, one can postpone it when in rush, but make it show up as a CI faiure on the next working day. 
+- **remove** runs `npm rm` on the top level dependency containing the issue. It's a convenience option for the user to remove an old package which they no longer intend to use. 
+
+(this RFC can't define the full scope of investigate)
+- **investigate** when fix is not available, investigate option shows instead. It goes up through the dependencies chain and finds the one that needs their package.json updated with new version specification to enable a fix. 
+The result can be a call to action to create a PR (patch could be automatically generated)
+Patch could also be applied locally or a specific change to package-lock.json could be proposed. 
+
+**skip** and **abort** options should also be provided.
+
+## Rationale and Alternatives
+
+**Ignoring is necessary but must be under control at all times**
+- `nsp check` used to support ignoring a particular issue in all dependencies, but that's not enough to be in control and remain secure. It wasn't uncommon for projects to ignore an advisory id because it was coming up as an issue in code that didn't run in production, thus risking it getting into the production code when updating another dependency to a version with the same issue.
+- ignoring just dev dependencies is not enough. Sometimes the developer knows a vulnerability can be ignored because a package with a DOS is used in tooling for the project (not a dev dependency but eg. a migration script), not production code. To be really secure, ignoring must be explicit. 
+- Editing a file to specify what to ignore will not suffice any more at that level of detail, so automation is necessary.
+
+**Why audit-resolv.json?**
+Resolutions must be recorded in a file and committed to git(or alternative) for edit history and full audit on who decided what.
+Resolution format must be readable for a JavaScript developer.
+
+- `package-lock.json` is not a good option for storing resolutions because even if we overlook the fact that it'd be adding another purpose to a single-purpose file, the community at large is still used to removing and regenerating it often. 
+- `package.json` is not a good option, because it is already overloaded for so many functionalities. The output of resolutions could be lengthy and clutter the file. User will need the resolution information much less often than they look in the package.json file. ALso, resolution file is not meant to be edited by hand.
+- `.npmrc` is not a good option because it doesn't live in the repository and being a dotfile is easy to miss, so a developer could overlook it affecting the audit. Also using it for the purpose of storing resolutions to project specific issues seems counter-intuitive.
```

Comment:
npmrc *could* live in the repo, it just also might not.

Reply:
I'm aware of that, but we probably wouldn't want to make an exception and accept global resolutions too. That could lead to hard to debug problems.

Reply:
Totally agree; just hoping to clarify the npmrc explanation.